preload_qsos_dr7.m

```diff
@@ -50,26 +40,37 @@
 	% Here file_loader uses dr7 spectrum reader function and given mpf to read
 	% spectrum 
   
-  % % % Masking Sky lines 
+  % bit 1: the spectrum with median SN<4 should be filtered
+  if (nanmedian(this_flux./sqrt(this_noise_variance))<4)
+    filter_flags(i) = bitset(filter_flags(i), 2, true);
+    continue;
+  end
+  % CIV removal region
+  % if (civ_rmove_in_train =true)
+    
+  %   maskCIV = 
+  %   this_pixel_mask()
+  % % Masking Sky lines 
+  if SkyLine==1
+    this_pixel_mask((abs(this_wavelengths-5579)<5) & (abs(this_wavelengths-6302)<5))=1;
+    this_pixel_mask((this_wavelengths>6868) & (this_wavelengths<6932))=1;
+    this_pixel_mask((this_wavelengths>7594) & (this_wavelengths<7700))=1;
+  end
 
-  % this_pixel_mask((abs(this_wavelengths-5579)<5) & (abs(this_wavelengths-6302)<5))=1;
-  % this_pixel_mask((this_wavelengths>6868) & (this_wavelengths<6932))=1;
-  % this_pixel_mask((this_wavelengths>7594) & (this_wavelengths<7700))=1;
-  this_rest_wavelengths = emitted_wavelengths(this_wavelengths, all_zqso(i));
+  % masking pixels with SN<4
+  this_pixel_mask(this_flux./sqrt(this_noise_variance)<4)=1;
```

Comment:
do you need to specify you are assigning True values to a Boolean array?

Reply:
I think this line should be deleted. I don't need this filtering. It was a mistake.